odoo/models.py

```diff
@@ -1698,6 +1699,97 @@ def _read_group_fill_results(self, domain, groupby, remaining_groupbys,
 
         return list(result.values())
 
+    @api.model
+    def _read_group_fill_temporal(self, data, groupby, aggregated_fields,
+                                  interval=dateutil.relativedelta.relativedelta(months=1)):
+        """Helper method for filling dates 'holes' in a result set.
+
+        We are in a use case where data are grouped by a date field (typically
+        months but it could a another interval) and displayed in a Bar Chart or
+        a Line Chart.
+
+        Let's says a request has to group records by month for August, September
+        and ...December.  If we don't changed anything, we would get a Bar Chart
+        looking like this :
+                                                ___
+                                      ___      |   |
+                                     |   |     |   |
+                                     |   | ___ |   |
+                                     |   ||   ||   |
+                                     |___||___||___|
+                                      Aug  Sep  Dec
+
+        December follows directly after September, it can be unintuitiv for the
+        user, so we change that. We add some fake records for each missing months
+        between the earliest and the lastest date of the result
```

Comment:
The problem is that December data follows immediately September data, which is misleading for the user. Adding explicit zeroes for missing data gives something like